cortex/secondary/call_duration.py

```diff
@@ -27,15 +29,28 @@ def call_duration(incoming=True, **kwargs):
             timestamp (int): The beginning of the window (same as kwargs['start']).
             value (float): The time spent in a call.
     """
-    incoming_dict = {True: 1, False: 2}
-    label = incoming_dict[incoming]
-    log.info('Loading raw calls data...')
-    _calls = calls(**kwargs)['data']
-    log.info('Computing call duration...')
-    _call_duration = np.sum([call['call_duration'] for call in _calls
-                             if call['call_type'] == label])
+    log.info('Computing call duration ...')
+
+    _calls = telephony(**kwargs)['data']
+
+    call_duration = np.sum(call['duration'] for call in _calls)
+
+    if incoming == "all":
+        call_duration = np.sum(call['duration'] for call in _calls
+                               if (call['type'] == "incoming"
+                                   or call['type'] == "outgoing"))
+
+    if incoming == "incoming":
+        call_duration = np.sum(call['duration'] for call in _calls
+                               if call['type'] == "incoming")
+
+    if incoming == "outgoing":
+        call_duration = np.sum(call['duration'] for call in _calls
+                               if call['type'] == "outgoing")
+
```

Comment:
This is more just a thought but it might be worth adding (at least) a warning here if `incoming` isn't passed one of 'all', 'incoming', or 'outgoing'. I think having the default behavior be to return all call duration data is correct but if someone makes a typo and accidentally types `outgoig` in their function some sort of warning should be thrown.